flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -130,4 +119,42 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
             return new StaticFileStoreSplitEnumerator(context, snapshot, splits);
         }
     }
+
+    private DataFilePlan batchScanSplits(DataTableScan scan) {
+        return scan.plan();
+    }
+
+    private DataFilePlan continuousFirstScanSplits(
+            DataTableScan scan, SnapshotManager snapshotManager) {
+        switch (startupMode) {
```

Comment:
Is it better to introduce `DataFilePlanBuilder` to create `DataFilePlan` for different startup mode?

Reply:
I think we can extract this logical to `SnapshotEnumerator.startup`.
We don't need to introduce `DataFilePlanBuilder`, it is just a simple call.